best-practices-zarr-ndarray.md

```diff
@@ -0,0 +1,536 @@
+          "unit": "K"
+        },
+        {
+          "name": "ta",
+          "description": "air temperature",
+          "cf:standard_name": "air_temperature",
+          "unit": "K"
+        }
+      ]
+    }
+  }
+}
+```
+
+**Key Points**:
+- Single asset represents an entire xarray Dataset
+- Multiple dimension types: temporal, spatial (non-standard grid), vertical
+- Variables have different dimensional structures (2D vs 3D)
+- CF standard names are preserved using a `cf:` prefix
+- The `bands` array provides human-readable variable descriptions
+- Non-geographic spatial reference system (HEALPix) is indicated
+
+### Virtual Zarr with Reference Files
+
+**Context**: Kerchunk and similar tools create virtual Zarr stores that reference existing data files (e.g., NetCDF) without copying data. This enables Zarr-like access to legacy formats.
+
+**STAC Representation**:
+
+```json
+{
+  "type": "Feature",
+  "stac_version": "1.1.0",
+  "id": "CMIP6_ScenarioMIP_NCAR_CESM2",
+  "properties": {
+    "start_datetime": "2015-01-01T00:00:00Z",
+    "end_datetime": "2024-12-31T23:59:59Z"
+  },
+  "links": [
+    {
+      "rel": "store",
+      "href": "https://storage.example.com/kerchunk/CMIP6_CESM2_rlus_kr1.0.json",
+      "type": "application/json+zstd",
+      "title": "Kerchunk Reference File"
+    }
+  ],
+  "assets": {
+    "reference": {
+      "href": "https://storage.example.com/kerchunk/CMIP6_CESM2_rlus_kr1.0.json",
+      "type": "application/json+zstd",
+      "title": "Kerchunk reference file",
+      "roles": ["reference", "data"],
+      "description": "Virtual Zarr store referencing NetCDF files"
+    },
+    "source_data": {
+      "href": "https://storage.example.com/data/CMIP6/rlus_day_CESM2-WACCM_ssp585_gn_20150101-20241231.nc",
+      "type": "application/netcdf",
+      "title": "Original NetCDF file",
+      "roles": ["source"]
+    }
+  }
+}
+```
+
+**Key Points**:
+- The store link points to the reference file
+- Assets include both the reference file and source data
+- Media type indicates the reference format (e.g., JSON with zstd compression)
+- Role `"reference"` indicates virtual/indirect data access
+- Role `"source"` indicates the underlying data files
+
+## Asset Discovery and Access Patterns
+
+### Accessing Zarr Subgroups
+
+For hierarchical stores, provide assets at meaningful organizational levels:
+
+```json
+"assets": {
+  "measurements_r10m": {
+    "href": "s3://bucket/data.zarr/measurements/reflectance/r10m",
+    "roles": ["group", "data"],
+    "description": "10m resolution measurements"
+  },
+  "measurements_r20m": {
+    "href": "s3://bucket/data.zarr/measurements/reflectance/r20m",
+    "roles": ["group", "data"],
+    "description": "20m resolution measurements"
+  }
+}
+```
+
+### Multi-Resolution and Pyramid Data
+
+EARLY DRAFT: to be updated with multiscale conventions in Zarr.
+
+For data with multiple resolutions (pyramids), each resolution level should be a separate asset:
```

Comment:
So far that was our workaround for the `R10m/R20m/R60m` because there is no muli-scale concept in Zarr yet, but assuming Zarr supports multi-scales at some point I don't think if it's the best idea to model every resolution level as different assets. 

In the Sentinel sense multiscales over time kind of became different "products", but in the general sense they often are just a performance optimization for visualization (the classic COG pyramids for Dynamic WMS for example). In that case I don't think this approach is a good idea, as it will lead to lots of metadata duplication. Maybe an asset/band specific field could be used instead to indicate the available scales?